lib/AST/Decl.cpp

```diff
@@ -6404,6 +6404,16 @@ VarDecl *VarDecl::getPropertyWrapperWrappedValueVar() const {
   return getPropertyWrapperAuxiliaryVariables().localWrappedValueVar;
 }
 
+bool VarDecl::hasStorageOrWrapsStorage() const {
+  if (hasStorage())
+    return true;
+  auto *backing = getPropertyWrapperBackingProperty();
```

Comment:
Can you also add a check for getAttributes().hasAttribute<LazyAttr>()?

Reply:
I assume this suggestion is to avoid going into the request evaluator when it's not necessary. Should we instead add this check inside the `VarDecl::getPropertyWrapper*` methods to avoid doing it at all of the call sites?

Reply:
Sorry, a formatting error there -- I meant to write `getAttributes().hasAttribute<LazyAttr>()` -- we need the same check for lazy properties as well.

I'm assuming the short-circuiting check you're thinking of is for `CustomAttr`. I remember this was worthwhile at one point because we weren't caching the results of some of these requests, but I'm not sure anymore.

Reply:
Thanks, I've added the check for the `lazy` attribute as well.

Reply:
Whoops, yes, I thought you were talking about checking for custom attributes. Sorry!